### PR TITLE
EDSC-3665: Upgrade AWS SDK from v2 to v3

### PR DESCRIPTION
**Pull request by dpesall**
# Overview

### What is the feature?

Upgrading AWS SDK from v2 to v3

### What is the Solution?

In AWS SDK v3, services are modularized into individual service clients instead of one large AWS object.

### What areas of the application does this impact?

Implementations of AWS services such as SQS, S3, DynamoDB, SecretsManager

# Testing

### Reproduction steps

- Running the jest test suite locally

1. Pull down this branch of Earthdata Search
2. Run `npm run test`
3. Verify all tests pass

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: serverless/src/util/database/__tests__/getDbCredentials.test.js [hunk 1]
```diff
@@ -42,10 +47,9 @@ describe('getDbCredentials', () => {
...
-    expect(secretsManagerData.mock.calls[0]).toEqual([{
-      SecretId: 'test-DbPasswordSecret'
-    }])
+    expect(client.send).toBeCalledTimes(1)
+    expect(client.send).toHaveBeenCalledWith(
+      { SecretId: 'test-DbPasswordSecret' }
+    )
   })
-})
+})
```
Comment (eudoroolivares2016) [inline serverless/src/util/database/__tests__/getDbCredentials.test.js:55]:
```suggestion
})

```

File: serverless/src/util/__tests__/startOrderStatusUpdateWorkflow.test.js [hunk 2]
```diff
@@ -33,14 +38,16 @@ describe('startOrderStatusUpdateWorkflow', () => {
...
+        input: JSON.stringify({
+          id: 1,
+          accessToken: 'access-token',
+          orderType: 'ESI'
+        })
+      }
+    )
   })
-})
+})
```
Comment (eudoroolivares2016) [inline serverless/src/util/__tests__/startOrderStatusUpdateWorkflow.test.js:53]:
```suggestion
})

```

File: serverless/src/util/urs/getUrsSystemCredentials.js [hunk 3]
```diff
@@ -27,14 +27,12 @@ export const getUrsSystemCredentials = async (earthdataEnvironment) => {
...
 
-    console.log(`Fetching UrsSystemPasswordSecret_${earthdataEnvironment}`)
-
     const params = {
       SecretId: `UrsSystemPasswordSecret_${earthdataEnvironment}`
     }
 
     // If not running in development mode fetch secrets from AWS
-    const secretValue = await secretsmanager.getSecretValue(params).promise()
+    const secretValue = await secretsmanager.send(new GetSecretValueCommand(params))
```
Comment (eudoroolivares2016) [inline serverless/src/util/urs/getUrsSystemCredentials.js:35]:
```suggestion
    const command = new GetSecretValueCommand(params)
    const secretValue = await secretsmanager.send(command)
```

File: serverless/src/util/__tests__/getEdlConfig.test.js [hunk 4]
```diff
@@ -54,6 +57,7 @@ describe('getEdlConfig', () => {
       }
     })
 
-    expect(uatSecretsManagerData.promise).toBeCalledTimes(1)
+    expect(client.send).toBeCalledWith({ SecretId: 'UrsClientConfigSecret_uat' })
+    expect(client.send).toBeCalledTimes(2)
   })
-})
+})
```
Comment (eudoroolivares2016) [inline serverless/src/util/__tests__/getEdlConfig.test.js:63]:
```suggestion
})

```

File: serverless/src/util/__tests__/getAdminUsers.test.js [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
-    expect(secretsManagerData.mock.calls[0]).toEqual([{
-      SecretId: 'EDSC_Admins'
-    }])
+    expect(client.send).toBeCalledTimes(1)
+    expect(client.send).toHaveBeenCalledWith(
+      { SecretId: 'EDSC_Admins' }
+    )
   })
-})
+})
```
Comment (eudoroolivares2016) [inline serverless/src/util/__tests__/getAdminUsers.test.js:31]:
```suggestion
})

```

File: serverless/src/generateColorMaps/__tests/getProjectionCapabilities.test.js [hunk 6]
```diff
@@ -30,7 +25,6 @@ beforeEach(() => {
       debug: false
     })
 
-    // Mock the db connection
```
Comment (eudoroolivares2016) [inline serverless/src/generateColorMaps/__tests/getProjectionCapabilities.test.js:33]: Not sure if you intended to delete this comment. I think this mockKnex call is really self explanatory so its not really neded

Comment (dpesall) [reply serverless/src/generateColorMaps/__tests/getProjectionCapabilities.test.js:33]: Did not mean to delete that line, thank you for pointing it out!

**Review by jaybarra — APPROVED** [review]

**Pull request merged**